### PR TITLE
Added primary key output option

### PR DESCRIPTION
**Pull request by azhard**
- Updated google cloud dependancies
- Added output options to include list of primary key columns & list of primary keys in output of DML statements (updates and inserts)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -227,12 +227,17 @@
     <dependency>
       <groupId>com.google.cloud</groupId>
       <artifactId>google-cloud-pubsub</artifactId>
-      <version>0.30.0-beta</version>
+      <version>1.66.0</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>23.6-jre</version>
```
Comment (osheroff) [inline pom.xml]: why did we pick up guava here?

File: src/main/java/com/zendesk/maxwell/row/RowMap.java [hunk 2]
```diff
@@ -184,6 +185,27 @@ public String toJSON(MaxwellOutputConfig outputConfig) throws Exception {
...
+		EncryptionContext encryptionContext = null;
+		if (outputConfig.encryptionEnabled()) {
+			encryptionContext = EncryptionContext.create(outputConfig.secretKey);
+		}
+
+		DataJsonGenerator dataWriter = outputConfig.encryptionMode == EncryptionMode.ENCRYPT_DATA
+			? json.getEncryptingGenerator()
+			: json.getPlaintextGenerator();
+
+		JsonGenerator dataGenerator = dataWriter.begin();
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/row/RowMap.java]: this diff is weird, why did all this code move above there?

File: src/main/java/com/zendesk/maxwell/row/RowMap.java [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package com.zendesk.maxwell.row;
 
+import com.google.api.client.util.Lists;
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/row/RowMap.java]: not a fan of this dependency

File: src/test/java/com/zendesk/maxwell/row/RowMapTest.java [hunk 4]
```diff
@@ -51,7 +52,7 @@ public void testGetRowIdentity() {
 
 		RowMap rowMap = new RowMap("insert", "MyDatabase", "MyTable", TIMESTAMP_MILLISECONDS, pKeys, POSITION);
 
-		rowMap.putData("id", "9001");
+		rowMap.putData("id", 9001);
```
Comment (osheroff) [inline src/test/java/com/zendesk/maxwell/row/RowMapTest.java]: why should these tests change?  doesn't seem right to me...

**Comment by osheroff** [comment]
overall this is a fine feature and implemented dilligently enough (thx for doing documentation the first time!)

I'm happy to merge it after code review issues are addressed.

File: pom.xml (hunk 1 above)
Comment (azhard) [reply pom.xml]: Removed.

File: src/main/java/com/zendesk/maxwell/row/RowMap.java (hunk 2 above)
Comment (azhard) [reply src/main/java/com/zendesk/maxwell/row/RowMap.java]: We are using the dataWriter to write the primaryKeys and primaryKeyColumn objects but we need to write them before the excludeColumns logic happens. If we don't, any of the columns that make up the primary key that are also in the exlcudeColumns list will not be outputted.

File: src/main/java/com/zendesk/maxwell/row/RowMap.java (hunk 3 above)
Comment (azhard) [reply src/main/java/com/zendesk/maxwell/row/RowMap.java]: Removed.

File: src/test/java/com/zendesk/maxwell/row/RowMapTest.java (hunk 4 above)
Comment (azhard) [reply src/test/java/com/zendesk/maxwell/row/RowMapTest.java]: The test was changed to show how the "primary_key" is outputted when dealing with non string values. ie. "primary_key": [9001, "foo"] vs. "primary_key": ["9001", "foo"].

**Comment by azhard** [comment]
@osheroff addressed all your comments, let me know if anything else needs fixing!

File: docs/docs/dataformat.md [hunk 5]
```diff
@@ -30,6 +30,8 @@ mysql> insert into test.e set m = 4.2341, c = now(3), comment = 'I am a creature
    "position":"master.000006:800911",
    "server_id":23042,
    "thread_id":108,
+   "primary_key": "[1, "2016-10-21 05:33:37.523000"]"
```
Comment (osheroff) [inline docs/docs/dataformat.md]: these two aren't quite right but I'll fix after merge.

**Comment by osheroff** [comment]
nice work.  merging.

**Pull request merged**